cosmovisor/cmd/cosmovisor/init.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/rs/zerolog"
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/cosmovisor"
+	cverrors "github.com/cosmos/cosmos-sdk/cosmovisor/errors"
+)
+
+func init() {
+	rootCmd.AddCommand(initCmd)
+}
+
+var initCmd = &cobra.Command{
+	Use:          "init <path to executable>",
+	Short:        "Initializes a cosmovisor daemon home directory.",
+	Args:         cobra.ExactArgs(1),
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		logger := cmd.Context().Value(cosmovisor.LoggerKey).(*zerolog.Logger)
+
+		return InitializeCosmovisor(logger, args)
+	},
+}
+
+// InitializeCosmovisor initializes the cosmovisor directories, current link, and initial executable.
+func InitializeCosmovisor(logger *zerolog.Logger, args []string) error {
+	if len(args) < 1 || len(args[0]) == 0 {
+		return errors.New("no <path to executable> provided")
+	}
+	pathToExe := args[0]
+	switch exeInfo, err := os.Stat(pathToExe); {
+	case os.IsNotExist(err):
+		return fmt.Errorf("executable file not found: %w", err)
+	case err != nil:
+		return fmt.Errorf("could not stat executable: %w", err)
+	case exeInfo.IsDir():
+		return errors.New("invalid path to executable: must not be a directory")
+	}
+	cfg, err := getConfigForInitCmd()
+	if err != nil {
+		return err
+	}
+
+	logger.Info().Msg("Checking on the genesis/bin directory.")
+	genBinExe := cfg.GenesisBin()
+	genBinDir, _ := filepath.Split(genBinExe)
+	genBinDir = filepath.Clean(genBinDir)
+	switch genBinDirInfo, genBinDirErr := os.Stat(genBinDir); {
+	case os.IsNotExist(genBinDirErr):
+		logger.Info().Msgf("Creating directory (and any parents): %q", genBinDir)
```

Comment:
Nit, it appears that the logged messages starts with a lowercase in cosmovisor. Can you make it consistent (whichever casing you prefer)?